src/dotnet/Program.cs

```diff
@@ -157,6 +160,26 @@ internal static int ProcessArgs(string[] args, ITelemetry telemetryClient)
 
         }
 
+        private static void ConfigureDotNetForFirstTimeUse()
+        {
+            using (PerfTrace.Current.CaptureTiming())
+            {
+                using (var nugetPackagesArchiver = new NuGetPackagesArchiver())
```

Comment:
What would you suggest? Not having the abstractions? Losing the testability of separation of concerns of the code? If you prove to me that this actually has any performance impact in the overall run of dotnet (restore, new, build, etc) I will tune it. But given that the majority of time in dotnet is spent in other things, I prefer to keep the qualify of the code.